### PR TITLE
Implement menu component & Auth related fixes

### PR DESCRIPTION
**Pull request by kurund**
## Summary

* Implement new menu component
* Clean up existing code to use the same.

## Test Plan

* Included

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -31,7 +31,17 @@ describe('<App /> ', () => {
   });
 
   test('it should render <Chat /> component if session is active', async () => {
-    localStorage.setItem('session', '{"access_token":"access","renewal_token":"renew"}');
+    // let's create token expiry date for tomorrow
+    const tokenExpiryDate = new Date();
+    tokenExpiryDate.setDate(new Date().getDate() + 1);
+    console.log(new Date(tokenExpiryDate));
```
Comment (rathorevaibhav) [inline src/App.test.tsx:37]: remove

File: src/components/UI/Menu/MenuItem/MenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+export interface MenuItemProps {
+  title: string;
+  path: string;
+  onClickHandler: any;
+  className?: string;
+}
+
+const MenuItem: React.SFC<MenuItemProps> = (props) => {
+  let menuItemClass = '';
+  if (props.className === 'Warning') {
```
Comment (rathorevaibhav) [inline src/components/UI/Menu/MenuItem/MenuItem.tsx:15]: it's a minor thing but do you think we should rename to "danger", like other UI frameworks does?
btn-danger (red)
btn-warning (yellow)
btn-success (green)

**Review by rathorevaibhav — APPROVED** [review]

File: src/App.test.tsx (hunk 1 above)
Comment (kurund) [reply src/App.test.tsx]: Done.

File: src/components/UI/Menu/MenuItem/MenuItem.tsx (hunk 2 above)
Comment (kurund) [reply src/components/UI/Menu/MenuItem/MenuItem.tsx]: Since we are using objects, I have changed this to "Danger".

**Pull request merged**